### PR TITLE
Add SUDO_PIP environmental variable to enable pip caching on Travis

### PR DESCRIPTION
**Pull request by wkentaro**
This PR is created with following discussion in https://github.com/jsk-ros-pkg/jsk_travis/pull/295.
Close https://github.com/jsk-ros-pkg/jsk_travis/pull/295

For suggestion in https://github.com/jsk-ros-pkg/jsk_travis/pull/295#discussion_r74432237
@furushchev This is my proposing implementation, could you please give me some feedbacks?

### CODE REVIEW

**Comment by furushchev** [comment]
@wkentaro Sorry for late reply.
ok for me.

File: travis.sh [hunk 1]
```diff
@@ -75,8 +75,16 @@ echo "Testing branch $TRAVIS_BRANCH of $REPOSITORY_NAME"
 # Setup pip
 # FIXME: need to specify pip version to 6.0.7 to avoid unexpected error
 # https://github.com/jsk-ros-pkg/jsk_robot/pull/523#issuecomment-164699366
-sudo easy_install 'pip==6.0.7'
-sudo pip install -U -q pip setuptools
+if [ "$SUDO_PIP" != false ]; then
```
Comment (furushchev) [inline travis.sh]: @wkentaro Sorry, this part should be `if [ "$SUDO_PIP" = false ]; then`

Comment (wkentaro) [inline travis.sh]: Oops.. Thanks, I fixed it.

**Comment by wkentaro** [comment]
Please merge this.

**Comment by k-okada** [comment]
do we have any problems if we run `pip install` as non-root user for all cases?

**Comment by wkentaro** [comment]
> do we have any problems if we run pip install as non-root user for all cases?

I'm not sure. @furushchev Do you have some idea?

**Comment by furushchev** [comment]
@k-okada Sorry for late reply.
Generally, I think no.
Possible problem is when all pip packages are installed as non-root user and some command is mistakenly executed as root, it fails with no package error.

**Comment by furushchev** [comment]
@k-okada Is there any other problem?

**Comment by k-okada** [comment]
On Wed, Aug 31, 2016 at 12:49 PM, Furushchev <notifications@github.com> wrote:
>
> Possible problem is when all pip packages are installed as non-root user and some command is mistakenly executed as root, it fails with no package error.

If so, when do we have to set `SUDO_PIP=false`

--
◉ Kei Okada

**Comment by wkentaro** [comment]
What is this PR's status? Can I close this? @furushchev

**Comment by furushchev** [comment]
Sorry for delay, since we now use docker in travis, this is no longer
necessary.
I think it is ok to close.

-- ⌘ Yuki Furuta

2017-03-23 1:12 GMT+09:00 Kentaro Wada <notifications@github.com>:

**Comment by furushchev** [comment]
Thanks anyway :dango:

**Comment by wkentaro** [comment]
:smile:

**Pull request closed without merging**